framework/decode/openxr_replay_consumer_base.cpp

```diff
@@ -802,16 +832,32 @@ void OpenXrReplayConsumerBase::Process_xrPollEvent(const ApiCallInfo&
     XrInstance         in_instance = MapHandle<OpenXrInstanceInfo>(instance, &CommonObjectInfoTable::GetXrInstanceInfo);
     XrEventDataBuffer* capture_event = eventData->GetPointer();
 
-    // We received events that haven't been handled yet already, so see if this one is in the list already
-    for (size_t ii = 0; ii < received_events_.size(); ++ii)
+    // If we have already received some events, check this event against one of the previously
+    // occurring events.  If it has already occurred, remove it from the list and return.
+    for (size_t ii = 0; ii < unhandled_events_from_replay_.size(); ++ii)
     {
-        if (received_events_[ii].type == capture_event->type)
+        if (unhandled_events_from_replay_[ii].type == capture_event->type)
         {
             GFXRECON_LOG_WARNING("Previously received event %s (0x%x, %u)",
                                  GetEventTypeString(capture_event->type),
                                  capture_event->type,
                                  capture_event->type);
-            received_events_.erase(received_events_.begin() + ii);
+            unhandled_events_from_replay_.erase(unhandled_events_from_replay_.begin() + ii);
```

Comment:
INFO instead of WARNING?

Reply:
Also, why are we erasing everything in `unhandled...` before this one?

Reply:
I don't follow.  It's the beginning + index (or ii) which should be the element.  Why would it be erasing everything before this one?